### PR TITLE
Optimize big int circuits to use base 2^85

### PR DESCRIPTION
**Pull request by rahulghangas**
As title

### CODE REVIEW

File: circuits/batchverify.circom [hunk 1]
```diff
@@ -14,8 +14,8 @@ template BatchVerify(n, m) {
   signal input R8[m][256];
   signal input S[m][255];
 
-  signal input PointA[m][4][5];
-  signal input PointR[m][4][5];
+  signal input PointA[m][4][3];
+  signal input PointR[m][4][3];
```
Comment (jinankjain) [inline circuits/batchverify.circom:18]: I think this should be parameterized as well.

```
var ed25519_power = 255;
var base = 85;
var arr_size = ed25519_power\base;
```

File: circuits/modinv.circom [hunk 2]
```diff
@@ -6,34 +6,34 @@ include "./utils.circom";
 include "./lt.circom";
 
 template BigModInv51() {
```
Comment (jinankjain) [inline circuits/modinv.circom:8]: let's make this generalized and pass a base.

File: circuits/modinv.circom [hunk 3]
```diff
@@ -6,34 +6,34 @@ include "./utils.circom";
...
-  signal input in[5];
-  signal output out[5];
+  signal input in[3];
+  signal output out[3];
 
-  var p[5] = [2251799813685229, 2251799813685247, 2251799813685247, 2251799813685247, 2251799813685247];
+  var p[3] = [38685626227668133590597613, 38685626227668133590597631, 38685626227668133590597631];
```
Comment (jinankjain) [inline circuits/modinv.circom:12]: these constants can be 2 ** base;

File: circuits/modinv.circom [hunk 4]
```diff
@@ -6,34 +6,34 @@ include "./utils.circom";
...
   }
-  component mod = ModulusWith25519Chunked51(10);
-  for (var i = 0; i < 10; i++) {
+  component mod = ModulusWith25519Chunked51(6);
+  for (var i = 0; i < 6; i++) {
     mod.in[i] <== mult.out[i];
   }
   mod.out[0] === 1;
-  for (var i = 1; i < 5; i++) {
+  for (var i = 1; i < 3; i++) {
```
Comment (jinankjain) [inline circuits/modinv.circom:36]: I think all these constants can be base dependent.

File: circuits/modulus.circom [hunk 5]
```diff
@@ -338,84 +338,84 @@ template ModulusAgainst2Q() {
 
 template ModulusWith25519Chunked51(n) {
```
Comment (jinankjain) [inline circuits/modulus.circom:339]: Here also we can pass a base and then make rest of the constants dependent on that base.

**Comment by rahulghangas** [comment]
@jinankjain this will require quite a refactor. Also some design discussion needed, some things should not be generalized because they're ed25519 specific. I'm going to merge this branch and open an issue to track generalization of circuits

**Pull request merged**